### DISCUSSION TITLE
How to pass a configuration file to the language server?

### DISCUSSION THREAD

**Discussion by InSyncWithFoo** [microsoft/pyright#7650] [Q&A] [answered]
[This page](https://microsoft.github.io/pyright/#/settings) says, multiple times, that a setting can be overridden by the configuration file. So how do I pass the configuration file content or path to the language server? Or how is this setting inferred from other information?

**Comment by erictraut** [comment, marked as answer]
You place your configuration file (either a `pyrightconfig.json` or `pyproject.toml`) in the root directory of your project. Typically, you'll want to commit your config file to the repo so it can be shared by all of your project's collaborators. By contrast, language server settings are typically not committed to the repo and are specific to each developer.

**Reply by InSyncWithFoo** [reply]
The CLI has a way to define a path, however: `--project`. Is this difference deliberate?

**Reply by erictraut** [reply]
Yes, this is deliberate. Different use cases.

**Reply by fkrauthan** [reply]
Is there a reason why this is not supported. My use-case is that we have a monorepo with lets call it team specific project folders. Each team has slightly different `pyright` settings.
